### PR TITLE
Regform reorg

### PR DESCRIPTION
**Pull request by emcelroy**

### CODE REVIEW

**Comment by emcelroy** [comment]
@scytacki I decided to stop short of a full update to the new code style in these. I think it'll take me too much time unfortunately.

One question though: The changes we made regarding Formsy resulted in a "Formsy is not defined" error even though the form worked as expected. An easy fix was this change: https://github.com/concord-consortium/portal-pages/pull/91/commits/d40c488a6b171b1a534a445f7af5eb7b59a14be2 

I'm not sure that's the right thing to do though. Other globals defined in package.json don't seem to require that. So I think I'm missing a better option. If you have any suggestions on that, please let me know.

**Comment by scytacki** [comment]
@pjanik can you take a look at this PR, particularly this use of `global.Formsy`?

File: src/library/components/signup/basic_data_form.js [hunk 1]
```diff
@@ -1,35 +1,27 @@
...
+let PASS_TOO_SHORT
 
 PASS_TOO_SHORT = 'Password is too short'
 PASS_NOT_MATCH = 'Passwords do not match'
 INVALID_FIRST_NAME = 'Invalid first name. Use only letters and numbers.'
 INVALID_LAST_NAME = 'Invalid last name. Use only letters and numbers.'
 
-var TextInputClass = require('./text_input')
+const Formsy = global.Formsy
+const TextInputClass = require('./text_input')
```
Comment (pjanik) [inline src/library/components/signup/basic_data_form.js]: We tend to use `import something from './some-location'` form.

**Comment by emcelroy** [comment]
Thanks, @scytacki and @pjanik. I made that change from require to import, and fixed some small merge conflicts with master. Does it all look OK now?

File: src/library/components/common/external-report-button.js [hunk 2]
```diff
@@ -28,13 +28,13 @@ export default class ExternalReportButton extends React.Component {
   }
 
   componentWillReceiveProps (nextProps) {
-    this.setState({disabled: nextProps.isDisabled()})
```
Comment (scytacki) [inline src/library/components/common/external-report-button.js]: This looks wrong. @dougmartin  just changed this to:
`this.setState({disabled: nextProps.isDisabled()})`
I think @pjanik only changed some whitespace so I think the correct merge is:
`this.setState({ disabled: nextProps.isDisabled() })`

Comment (emcelroy) [reply src/library/components/common/external-report-button.js]: Ah, I see. Thanks for catching that. I fixed it in the latest commit.

**Pull request merged**